pylint/checkers/variables.py

```diff
@@ -1687,16 +1716,22 @@ def _check_consumer(
         if found_nodes is None:
             return (VariableVisitConsumerAction.CONTINUE, None)
         if not found_nodes:
-            if node.name in current_consumer.consumed_uncertain:
-                confidence = CONTROL_FLOW
-            else:
-                confidence = HIGH
-            self.add_message(
-                "used-before-assignment",
-                args=node.name,
-                node=node,
-                confidence=confidence,
-            )
+            if (
+                not self._postponed_evaluation_enabled
```

Comment:
This seems to be silencing checks for used-before-assignment having nothing to do with annotations, like this case, which used to raise a warning, and which I simplified from the latest primer results for sentry:

```python
from __future__ import annotations

if not annotations:
   asfd = 1

print(asfd)
```

Reply:
Looks great! See below for reasons I think we can postpone discussing those edge cases.

I think the last test we should add here would be one for this very issue with `from __future__ import annotations`, then I'd be glad to merge!